boring/src/ssl/test/session_resumption.rs

```diff
@@ -0,0 +1,159 @@
+use super::server::Server;
+use crate::ssl::test::MessageDigest;
+use crate::ssl::SslRef;
+use crate::ssl::SslSession;
+use crate::ssl::SslSessionCacheMode;
+use crate::ssl::TicketKeyCallbackResult;
+use crate::symm::Cipher;
+use std::ffi::c_void;
+use std::sync::atomic::{AtomicU8, Ordering};
+use std::sync::OnceLock;
+
+static CUSTOM_ENCRYPTION_CALLED_BACK: AtomicU8 = AtomicU8::new(0);
+static CUSTOM_DECRYPTION_CALLED_BACK: AtomicU8 = AtomicU8::new(0);
+
+#[test]
+fn resume_session() {
+    static SESSION_TICKET: OnceLock<Vec<u8>> = OnceLock::new();
+
+    let mut server = Server::builder();
+    server.expected_connections_count(2);
+    let server = server.build();
+
+    let mut client = server.client();
+    client
+        .ctx()
+        .set_session_cache_mode(SslSessionCacheMode::CLIENT);
+    client.ctx().set_new_session_callback(|_, session| {
+        let _can_receive_multiple_tickets = SESSION_TICKET.set(session.to_der().unwrap());
```

Comment:
```suggestion
        SESSION_TICKET.set(session.to_der().unwrap());
```

Reply:
`OnceLock:set()` returns a Result which we need to consume. I'll refactor to add a comment instead.

> Returns Ok(()) if the cell's value was set by this call.

Reply:
You can just unwrap it since this is tests

Reply:
Since we receive 2 NST from the server, [OnceLock::set](https://doc.rust-lang.org/beta/std/sync/struct.OnceLock.html#method.set).unwrap() will succeed for the 1st ticket but fail for the 2nd one. Since I just care to retrieve any NST, I need to ignore the result here.

Alternatively, I could do a `OnceLock::get().is_some()` check before `set`ting the value, which would avoid an Error value.

Reply:
I update the logic and added checks for the expected behavior.